orttraining/orttraining/test/training_api/common/synthetic_data_loader.cc

```diff
@@ -38,32 +40,54 @@ void RandomInts(std::vector<IntType>& rets, IntType low, IntType high) {
 
 }  // namespace
 
+template <typename T>
+void SyntheticSampleBatch::AddIntInput(const std::vector<int64_t>& shape, T low, T high) {
+  data_vector_.emplace_back(SyntheticInput(shape));
+
+  std::vector<T> values(data_vector_.back().NumOfElements());
+  RandomInts(values, low, high);
+
+  SyntheticDataVector& data = data_vector_.back().GetData();
+  data = values;
+  input_count_++;
+}
+
 void SyntheticSampleBatch::AddInt64Input(const std::vector<int64_t>& shape, int64_t low, int64_t high) {
-  data_vector_.emplace_back(std::make_unique<TypedSyntheticInput<int64_t>>(shape));
-  RandomInts(data_vector_.back()->GetData<int64_t>(), low, high);
+  AddIntInput(shape, low, high);
 }
 
 void SyntheticSampleBatch::AddInt32Input(const std::vector<int64_t>& shape, int32_t low, int32_t high) {
-  data_vector_.emplace_back(std::make_unique<TypedSyntheticInput<int32_t>>(shape));
-  RandomInts(data_vector_.back()->GetData<int32_t>(), low, high);
+  AddIntInput(shape, low, high);
+}
+
+void SyntheticSampleBatch::AddBoolInput(const std::vector<int64_t>& shape) {
+  // Use uint8_t to store the bool value by intention, because vector<bool> is specialized, we can not create a
+  // Tensor leveraging C APIs to reuse the data buffer.
+  AddIntInput(shape, static_cast<uint8_t>(0), static_cast<uint8_t>(1));
 }
 
 void SyntheticSampleBatch::AddFloatInput(const std::vector<int64_t>& shape) {
-  data_vector_.emplace_back(std::make_unique<TypedSyntheticInput<float>>(shape));
-  RandomFloats(data_vector_.back()->GetData<float>());
+  data_vector_.emplace_back(SyntheticInput(shape));
+
+  std::vector<float> values(data_vector_.back().NumOfElements());
```

Comment:
>(data_vector_.back().NumOfElements( [](http://example.com/codeflow?start=27&length=35)

Same thing here. Initializing with zeros that are likely thrown away later.

Reply:
Fixed